iavl/iavl_node.go

```diff
@@ -291,10 +417,14 @@ func (node *IAVLNode) set(t *IAVLTree, key []byte, value []byte) (newSelf *IAVLN
 			node.rightHash = nil // rightHash is yet unknown
 		}
 		if updated {
+			node.validate(t)
 			return node, updated
 		} else {
 			node.calcHeightAndSize(t)
-			return node.balance(t), updated
+			root := node.balance(t)
+			root.validate(t)
```

Comment:
Sorry for the dumb question, but why we're validating after each step? Is it because we are not sure in the impl and do not have enough tests?